lang/cs/Org.Apache.REEF.Network/Group/Operators/IOperatorTopology.cs

```diff
@@ -0,0 +1,114 @@
+using Org.Apache.REEF.Network.Group.Operators.Impl;
+using Org.Apache.REEF.Network.Group.Task.Impl;
+using Org.Apache.REEF.Network.NetworkService;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Exceptions;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Remote;
+
+namespace Org.Apache.REEF.Network.Group.Task
+{
+    /// <summary>
+    /// Contains the Operator's topology graph.
+    /// Used to send or receive messages to/from operators in the same
+    /// Communication Group.
+    /// </summary>
+    /// <typeparam name="T">The message type</typeparam>
+    public interface IOperatorTopology<T>
+    {
+        /// <summary>
+        /// Sends the message to the parent Task.
+        /// </summary>
+        /// <param name="message">The message to send</param>
+        /// <param name="type">The message type</param>
+        void SendToParent(T message, MessageType type);
+
+        /// <summary>
+        /// Sends the message to all child Tasks.
+        /// </summary>
+        /// <param name="message">The message to send</param>
+        /// <param name="type">The message type</param>
+        void SendToChildren(T message, MessageType type);
+
+        /// <summary>
+        /// Splits the list of messages up evenly and sends each sublist
+        /// to the child Tasks.
+        /// </summary>
+        /// <param name="messages">The list of messages to scatter</param>
+        /// <param name="type">The message type</param>
+        void ScatterToChildren(List<T> messages, MessageType type);
+
+        /// <summary>
+        /// Splits the list of messages up into chunks of the specified size 
+        /// and sends each sublist to the child Tasks.
+        /// </summary>
+        /// <param name="messages">The list of messages to scatter</param>
+        /// <param name="count">The size of each sublist</param>
+        /// <param name="type">The message type</param>
+        void ScatterToChildren(List<T> messages, int count, MessageType type);
+
+        /// <summary>
+        /// Splits the list of messages up into chunks of the specified size 
+        /// and sends each sublist to the child Tasks in the specified order.
+        /// </summary>
+        /// <param name="messages">The list of messages to scatter</param>
+        /// <param name="order">The order to send messages</param>
+        /// <param name="type">The message type</param>
+        void ScatterToChildren(List<T> messages, List<string> order, MessageType type);
+
+        /// <summary>
+        /// Receive an incoming message from the parent Task.
+        /// </summary>
+        /// <returns>The parent Task's message</returns>
+        T ReceiveFromParent();
+
+        /// <summary>
+        /// Receive a list of incoming messages from the parent Task.
+        /// </summary>
+        /// <returns>The parent Task's list of messages</returns>
+        List<T> ReceiveListFromParent();
```

Comment:
done everywhere...in once place i have to change iList to List within the code to call GetRange function